File: JS/面试经典150题/数组、字符串/88_合并两个有序数组.js
var merge = function(nums1, m, nums2, n) {
    let index1 = m - 1, index2 = n -1
    let mergeIndex = m + n - 1
    while(index2 >= 0){
        nums1[mergeIndex] = nums1[index1]>nums2[index2] ? nums1[index1--] : nums2[index2--]
        mergeIndex--
    }
    return nums1
};

File: Python/面试经典150题/数组、目录/88_合并两个有序数组.py
from typing import List
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p1, p2, now = m-1, n-1, m + n - 1
        while p2 >= 0:
            if p1>=0 and nums1[p1] > nums2[p2]:
                nums1[now] = nums1[p1]
                p1 -= 1
            else:
                nums1[now] = nums2[p2]
                p2 -= 1
            now -= 1
        print(nums1)

nums1 = [2,0]
nums2 = [1]
Solution().merge(nums1, 1, nums2, 1)